python/ray/state.py

```diff
@@ -277,6 +277,16 @@ def job_table(self):
 
         return results
 
+    def next_job_id(self):
```

Comment:
Why two `next_job_id` definitions?

Reply:
One is in state, another is in `GlobalState` used by state.

Reply:
Not all methods in state.py have two definitions. Do you know when we need two and when we need only one?

Reply:
All methods except those related with placement group are first defined in `GlobalState` then referred in state.py. I believe the state.py is a (global) wrapper of `GlobalState` to separate the interfaces and implementations.
For placement group part I'd rather think it's a missing part.